### PR TITLE
add cid to Class

### PR DESCRIPTION
**Pull request by SauloSilva**
ref #2

### CODE REVIEW

File: src/backbone-metal.js [hunk 1]
```diff
@@ -88,6 +88,8 @@ function _wrapAll(dest, source) {
  * @memberOf Backbone
  */
 var Class = Metal.Class = function() {
+  this.cidPrefix = this.cidPrefix ? this.cidPrefix : 'bmc';
+  this.cid = _.uniqueId(this.cidPrefix);
```
Comment (jasonLaster) [inline src/backbone-metal.js:91]: Can we add cidPrefix to the prototype?

Comment (SauloSilva) [inline src/backbone-metal.js:91]: I think be better

**Comment by jasonLaster** [comment]
:+1:

File: src/backbone-metal.js [hunk 2]
```diff
@@ -88,10 +88,18 @@ function _wrapAll(dest, source) {
...
   this.initialize(...arguments);
 };
 
 /**
+ * Identifier prefix of the cid. It can be override.
+ * @public
+ * @attributeOf Class {String}
+ */
+Class.prototype.cidPrefix = 'bmc';
```
Comment (jamiebuilds) [inline src/backbone-metal.js]: `metal` is probably a better one here, even I had to think about what `bmc` meant

File: test/class.js [hunk 3]
```diff
@@ -11,6 +11,38 @@ describe('Class', function() {
...
+    });
+  });
+
+  describe('#cidPrefix', function() {
+    beforeEach(function() {
+      this.prefix = 'subClass';
+
+      var Subclass = Metal.Class.extend({
+        property1: 'value1',
+        method1: this.method1,
```
Comment (jamiebuilds) [inline test/class.js]: what's the point of `property1` and `method1` in here?

File: test/class.js [hunk 4]
```diff
@@ -11,6 +11,38 @@ describe('Class', function() {
     });
   });
 
+  describe('#cid', function() {
+    beforeEach(function() {
+      this.instance = new Metal.Class();
+    });
+
+    it('should contains cid', function() {
+      expect(this.instance).to.include.keys('cid');
```
Comment (jamiebuilds) [inline test/class.js]: Could you assert what `cid` is? Something like:

``` js
expect(this.instance).to.have.property('cid').that.match(/^metal[0-9]+$/)
```

Also another test that asserts that `cid`'s are different.

**Comment by paulfalgout** [comment]
squash the commits?

**Comment by SauloSilva** [comment]
@paulfalgout yes, done.

**Comment by jasonLaster** [comment]
nice work. @SauloSilva you're the cid guy

**Comment by SauloSilva** [comment]
thanks @jasonLaster :smile:

**Pull request merged**